src/microbe_stage/PatchRegion.cs

```diff
@@ -0,0 +1,304 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   A region is a something like a continent/ocean that contains multiple biomes(patches).
+/// </summary>
+[UseThriveSerializer]
+public class PatchRegion
+{
+    [JsonProperty]
+    public readonly int ID;
+
+    [JsonIgnore]
+    public readonly ISet<PatchRegion> Adjacent = new HashSet<PatchRegion>();
+
+    [JsonIgnore]
+    public float PatchNodeWidth = 64.0f;
+
+    [JsonIgnore]
+    public float PatchNodeHeight = 64.0f;
+
+    [JsonIgnore]
+    public float RegionLineWidth = 4f;
+
+    [JsonIgnore]
+    public float PatchMargin = 4f;
+
+    [JsonProperty]
+    public string RegionType;
+
+    [JsonProperty]
+    public float Height;
+
+    [JsonProperty]
+    public float Width;
+
+    [JsonProperty]
+    public List<Patch> Patches;
+
+    [JsonConstructor]
+    public PatchRegion(int id, LocalizedString name, string regionType, Vector2 screenCoordinates,
+        float height, float width, List<Patch> patches)
+    {
+        ID = id;
+        Name = name;
+        RegionType = regionType;
+        Patches = patches;
+        ScreenCoordinates = screenCoordinates;
+        Height = height;
+        Width = width;
+    }
+
+    public PatchRegion(int id, LocalizedString name, string regionType, Vector2 screenCoordinates)
+    {
+        ID = id;
+        Patches = new List<Patch>();
+        Name = name;
+        Height = 0;
+        Width = 0;
+        RegionType = regionType;
+        ScreenCoordinates = screenCoordinates;
+    }
+
+    [JsonProperty]
+    public LocalizedString Name { get; private set; }
+
+    /// <summary>
+    ///   Coordinates this region is to be displayed in the GUI
+    /// </summary>
+    [JsonProperty]
+    public Vector2 ScreenCoordinates { get; set; }
+
+    public void BuildPatches(Random random)
```

Comment:
This looks like patch generation code, so should be moved to the actual patch generator.